src/ring_theory/subring/basic.lean

```diff
@@ -434,6 +434,16 @@ instance : has_top (subring R) :=
 
 @[simp] lemma coe_top : ((⊤ : subring R) : set R) = set.univ := rfl
 
+/-- The ring equiv between the top element of `subring R` and `R`. -/
+@[simps]
+def top_equiv : (⊤ : subring R) ≃+* R :=
```

Comment:
Can you move this to `subsemiring`?

Reply:
I moved this proof to `subsemiring` and used it to prove the versions for `subring` and `subfield`